### PR TITLE
doc: Mention Windows XP end of support in release notes

### LINKED ISSUE #7681: End support for Windows XP and older
Drop support for Windows XP and older - at some point. 

Microsoft support for Windows XP ended April 8th, 2014 (https://www.microsoft.com/en-us/WindowsForBusiness/end-of-xp-support). This means that not even critical security updates are being released anymore. This makes it dangerous to run a wallet on this OS.

There are also some strange, apparently random crashes on WinXP with 0.12, see #7674 and #7639.

I haven't heard of any reports of this happening on other versions of windows.

I hope it is possible to debug and fix this, but if not, and it turns out to be truly a WinXP only issue, we'll have to announce this through the appropriate channels.

### PR DESCRIPTION
**Pull request by laanwj**
Closes #7681, part of #7678

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -16,6 +16,28 @@ To receive security and update notifications, please subscribe to:
...
+Compatibility
+==============
+
+Microsoft ended support for Windows XP on [April 8th, 2014](https://www.microsoft.com/en-us/WindowsForBusiness/end-of-xp-support),
+an OS initially released in 2001. This means that not even critical security
+updates will be released anymore. Without security updates, using a bitcoin
+wallet on a XP machine is irresponsible at least.
+
+In addition to that, with 0.12.x there have been varied reports of Bitcoin Core
+randomly crashing on Windows XP. It is [https://github.com/bitcoin/bitcoin/issues/7681#issuecomment-217439891](not clear)
```
Comment (MarcoFalke) [inline doc/release-notes.md]: Nit: Format is `[name](http://foo.bar)`

Comment (laanwj) [inline doc/release-notes.md]: Yes will fix that, thanks, always get those swapped

File: doc/release-notes.md [hunk 2]
```diff
@@ -16,6 +16,28 @@ To receive security and update notifications, please subscribe to:
...
+what the source of these crashes is, but it is likely that upstream
```
Comment (fanquake) [inline doc/release-notes.md]: nit s/is/are

**Comment by fanquake** [comment]
ACK post nits being fixed.

**Comment by paveljanik** [comment]
ACK post nits being fixed.

File: doc/release-notes.md (hunk 2 above)
Comment (sipa) [inline doc/release-notes.md]: @fanquake 'is' looks correct here. The source is singular.

**Pull request merged**